### PR TITLE
chore: fix dependency convergence: javassist

### PR DESCRIPTION
**Pull request by ZheSun88**
```
[ERROR] Dependency convergence error for org.javassist:javassist:jar:3.28.0-GA paths to dependency are:
[ERROR] +-com.example:spring-skeleton:jar:1.0-SNAPSHOT
[ERROR]   +-com.vaadin:vaadin-spring-boot-starter:jar:24.4.3:compile
[ERROR]     +-com.vaadin:vaadin-spring:jar:24.4.2:compile
[ERROR]       +-org.reflections:reflections:jar:0.10.2:compile
[ERROR]         +-org.javassist:javassist:jar:3.28.0-GA:compile
[ERROR] and
[ERROR] +-com.example:spring-skeleton:jar:1.0-SNAPSHOT
[ERROR]   +-com.vaadin:vaadin-spring-boot-starter:jar:24.4.3:compile
[ERROR]     +-com.vaadin:hilla:jar:24.4.1:compile
[ERROR]       +-com.vaadin:hilla-endpoint:jar:24.4.1:compile
[ERROR]         +-com.github.javaparser:javaparser-symbol-solver-core:jar:3.25.10:compile
[ERROR]           +-org.javassist:javassist:jar:3.30.2-GA:compile

```

### CODE REVIEW

File: vaadin-spring/pom.xml [hunk 1]
```diff
@@ -173,9 +173,18 @@
             <groupId>org.reflections</groupId>
             <artifactId>reflections</artifactId>
             <version>0.10.2</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.javassist</groupId>
+                    <artifactId>javassist</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
```
Comment (mcollovati) [inline vaadin-spring/pom.xml:182]: If we pin the `javassist` dependency version, the exclusion should not be required IIRC

Comment (ZheSun88) [reply vaadin-spring/pom.xml:182]: oh.. okay, i will remove that exclusion then..

Comment (ZheSun88) [reply vaadin-spring/pom.xml]: oops.. 
 build failed https://github.com/vaadin/flow/actions/runs/9581065847/job/26417027620?pr=19607#step:13:103

```
Error:  Dependency convergence error for org.javassist:javassist:jar:3.28.0-GA paths to dependency are:
Error:  +-com.vaadin:vaadin-spring:jar:999.99-SNAPSHOT
Error:    +-org.reflections:reflections:jar:0.10.2:compile
Error:      +-org.javassist:javassist:jar:3.28.0-GA:compile
Error:  and
Error:  +-com.vaadin:vaadin-spring:jar:999.99-SNAPSHOT
Error:    +-org.javassist:javassist:jar:3.30.2-GA:compile
Error:  -> [Help 1]
```

Comment (mcollovati) [reply vaadin-spring/pom.xml:182]: Ok, so exclusion is still required.

**Review by mcollovati — APPROVED** [review]

**Pull request merged**